src/adapter/src/catalog.rs

```diff
@@ -2780,13 +2780,13 @@ impl Catalog {
                 cluster_replica_sizes: config.cluster_replica_sizes,
                 default_storage_cluster_size: config.default_storage_cluster_size,
                 availability_zones: config.availability_zones,
-                system_configuration: SystemVars::default(),
+                system_configuration: SystemVars::new(config.active_connection_count),
```

Comment:
When reading this value from the stash on bootstrap it should also be sync'd to this, otherwise all restarted envds will be at a 0 limit. I don't think even the LD sync would fix it because LD believes the current announced value to be correct, but it didn't get propagated here at boot.

Edit: ah, I see this was happening in SystemVars::new, more comments there.